src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -341,6 +341,13 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
                 ...(profile ? [new DirectoryMember(profile)] : []).map(toMemberResult),
                 ...publicRooms.map(toPublicRoomResult),
             ].filter(result => filter === null || result.filter.includes(filter));
+            // Filter out any IMemberResults for which we have a room
+            return results.filter((m) => !results.find((r) => {
+                const userId = m["member"]?.userId;
+                const roomId = r["room"]?.roomId;
+                if (!roomId || !userId) return false;
+                return DMRoomMap.shared().getUserIdForRoomId(roomId) === userId;
```

Comment:
1. this also filters out users if we’ve got a multi-user DM that started as a DM with that user
2. It’d be better to build a Set of all the userIds we don’t want to show up, then filter the results by membership of that set (which is O(n * log(n)), instead of using .find for every result (which is O(n²))

Reply:
Good points - thanks!